src/components/layout/IngredientForm.tsx

```diff
@@ -0,0 +1,161 @@
+
+    const deleteIngredient = useCallback((ingredient?: Ingredient) => {
+        if (ingredient !== undefined && ingredient.id !== undefined) {
+            ingredientService.deleteIngredient(ingredient.id)
+                .then(r => {
+                    createToast('success', `Ingredient ${ingredient.name} supprimé`);
+                    toggleModal();
+                })
+                .catch(e => createToast('error', e.response ? e.response.data.message : ""));
+        }
+    }, [createToast, toggleModal])
+
+    const formik = useFormik({
+        initialValues: {
+            name: !props.selectedIngredient ? '' : props.selectedIngredient.name,
+            price: !props.selectedIngredient ? undefined : props.selectedIngredient.price,
+            quantity: !props.selectedIngredient ? undefined : props.selectedIngredient.quantity,
+            type: !props.selectedIngredient ? IngredientType.BASE : props.selectedIngredient.type,
+            image: !props.selectedIngredient ? '' : props.selectedIngredient.image
+        },
+        onSubmit: values => {
+            const ingredientToSave: Ingredient = {
+                id: props.selectedIngredient ? props.selectedIngredient.id : '',
+                name: values.name,
+                price: values.price ? values.price : 0,
+                type: values.type,
+                quantity: !values.quantity ? 0 : values.quantity,
+                image: values.image,
+            }
+
+            const save = ingredientToSave.id ? updateIngredient(ingredientToSave) : addIngredient(ingredientToSave);
+
+            save.then(() => {
+                props.refreshIngredients()
+                formik.resetForm();
+            });
+        },
+        enableReinitialize: true
+    });
+
+    return (
+        <div>
+            <h3>{props.selectedIngredient ? "Mettre à jour un ingrédient" : "Créer un ingrédient" }</h3>
+            <div className="ingredient-form">
+                <form onSubmit={formik.handleSubmit}>
+                    <div className="ingredient-form-line">
+                        <input
+                            className="ingredient-form-input"
+                            id="name"
+                            name="name"
+                            type="text"
+                            placeholder="Nom"
+                            onChange={formik.handleChange}
+                            value={formik.values.name}
+                            required />
+                    </div>
+                    <div className="ingredient-form-line">
+                        <select className="ingredient-form-input"
+                            id="type"
+                            onChange={formik.handleChange}
+                            value={formik.values.type}
+                            placeholder="Catégorie"
+                            required>
+                            {
+                                ingredientTypeMapping.map((type) => (
```

Comment:
Parenthèses redondantes

```suggestion
ingredientTypeMapping.map(type => (
```